apps/server/src/institutions/institutionResolver.ts

```diff
@@ -35,72 +36,93 @@ const getAggregatorByVolume = (
     });
 
-  const aggregatorsWithFullSupport: Aggregator[] = getAvailableAggregators({
-    institution,
-    jobTypes,
-    shouldRequireFullSupport: true,
-    supportedAggregators: preferences.supportedAggregators,
-  });
-
-  const aggregators = aggregatorsWithFullSupport.length
-    ? aggregatorsWithFullSupport
-    : aggregatorsWithAtLeastPartialSupport;
-
-  let aggregator: Aggregator;
+    const aggregators = aggregatorsWithFullSupport.length
+      ? aggregatorsWithFullSupport
+      : aggregatorsWithAtLeastPartialSupport;
+
+    const potentialResolvers = [
+      () =>
+        getAggregatorByVolume(
+          preferences?.institutionAggregatorVolumeMap?.[ucpInstitutionId],
+        ),
+      () => getAggregatorByVolume(preferences?.defaultAggregatorVolume),
+      () => preferences?.defaultAggregator,
+    ];
+
+    for (const resolver of potentialResolvers) {
+      const possibleAggregator = resolver();
+
+      if (aggregators.includes(possibleAggregator)) {
+        aggregator = possibleAggregator;
+        break;
+      }
+    }
 
-  const potentialResolvers = [
-    () =>
-      getAggregatorByVolume(
-        preferences?.institutionAggregatorVolumeMap?.[institutionId],
-      ),
-    () => getAggregatorByVolume(preferences?.defaultAggregatorVolume),
-    () => preferences?.defaultAggregator,
-  ];
-
-  for (const resolver of potentialResolvers) {
-    const possibleAggregator = resolver();
-
-    if (aggregators.includes(possibleAggregator)) {
-      aggregator = possibleAggregator;
-      break;
+    if (!aggregator) {
+      aggregator = aggregators[Math.floor(Math.random() * aggregators.length)];
     }
-  }
 
-  if (!aggregator) {
-    aggregator = aggregators[Math.floor(Math.random() * aggregators.length)];
-  }
+    institutionAggregator = institution[
+      aggregator as keyof CachedInstitution
+    ] as InstitutionAggregator;
 
-  const institutionAggregator = institution[
-    aggregator as keyof CachedInstitution
-  ] as InstitutionAggregator;
+    const testAdapterName =
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (adapterMap[aggregator as keyof typeof adapterMap] as any)?.testAdapterId;
 
-  const testAdapterName =
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    (adapterMap[aggregator as keyof typeof adapterMap] as any)
-      ?.testInstitutionAdapterName;
+    if (testAdapterName && institution.is_test_bank) {
+      aggregator = testAdapterName;
+    }
 
-  if (testAdapterName && institution.is_test_bank) {
-    aggregator = testAdapterName;
+    debug(
+      `Resolving institution: ${ucpInstitutionId} to aggregator: ${aggregator} available aggregators: ${JSON.stringify(aggregators)}`,
+    );
   }
 
-  debug(
-    `Resolving institution: ${institutionId} to aggregator: ${aggregator} available aggregators: ${JSON.stringify(aggregators)}`,
-  );
-
   return {
     id: institutionAggregator?.id,
```

Comment:
If you add `oauth: institutionAggregator?.supports_oauth` to this object then I think you can use it in the getInstitutionsHandler